source/common/upstream/outlier_detection_impl.cc

```diff
@@ -94,16 +110,94 @@ Http::Code DetectorHostMonitorImpl::resultToHttpCode(Result result) {
   case Result::REQUEST_FAILED:
     http_code = Http::Code::InternalServerError;
     break;
-  case Result::SERVER_FAILURE:
-    http_code = Http::Code::ServiceUnavailable;
+  case Result::CONNECT_SUCCESS:
+    NOT_REACHED_GCOVR_EXCL_LINE; // SUCCESS is used to report connection level success, not
+                                 // transaction
     break;
   }
 
   return http_code;
 }
 
-void DetectorHostMonitorImpl::putResult(Result result) {
-  putHttpResponseCode(enumToInt(resultToHttpCode(result)));
+// Method is called by putResult when external and local origin errors
+// are not treated differently. All errors are mapped to HTTP codes.
+void DetectorHostMonitorImpl::putResultNoLocalExternalSplit(Result result,
+                                                            absl::optional<uint64_t> code) {
+  switch (result) {
+  case Result::CONNECT_SUCCESS:
+    // SUCCESS means that connection to host was established, but not transaction still can fail.
```

Comment:
I'm a little confused by this. Why don't we track this result type in the non-split case? Maybe it's related to my other comment. Can you potentially rephrase this comment? Maybe that will help.

Reply:
See my response to your comment titled: _Shouldn't this be CONNECT_SUCCESS?_
I will also update the comment.